TMessagesProj/src/main/java/org/telegram/messenger/PushListenerController.java

```diff
@@ -1472,4 +1477,59 @@ public boolean hasServices() {
             return hasServices;*/
         }
     }
+    public final static class UnifiedPushListenerServiceProvider implements IPushListenerServiceProvider {
+        public final static UnifiedPushListenerServiceProvider INSTANCE = new UnifiedPushListenerServiceProvider();
+
+        private UnifiedPushListenerServiceProvider(){};
+
+        @Override
+        public boolean hasServices() {
+            return !UnifiedPush.getDistributors(ApplicationLoader.applicationContext, new ArrayList()).isEmpty();
+        }
+
+        @Override
+        public String getLogTitle() {
+            return "UnifiedPush";
+        }
+
+        @Override
+        public void onRequestPushToken() {
+            String currentPushString = SharedConfig.pushString;
+            if (!TextUtils.isEmpty(currentPushString)) {
+                if (BuildVars.DEBUG_PRIVATE_VERSION && BuildVars.LOGS_ENABLED) {
+                    FileLog.d("UnifiedPush endpoint = " + currentPushString);
+                }
+            } else {
+                if (BuildVars.LOGS_ENABLED) {
+                    FileLog.d("No UnifiedPush string found");
+                }
+            }
+            Utilities.globalQueue.postRunnable(() -> {
+                try {
+                    SharedConfig.pushStringGetTimeStart = SystemClock.elapsedRealtime();
+                    SharedConfig.saveConfig();
+                    if (UnifiedPush.getDistributor(ApplicationLoader.applicationContext).isEmpty()) {
+                        List<String> distributors = UnifiedPush.getDistributors(ApplicationLoader.applicationContext, new ArrayList<>());
```

Comment:
```suggestion
                        List<String> distributors = UnifiedPush.getDistributors(ApplicationLoader.applicationContext, new ArrayList<String>(Collections.singleton(UnifiedPush.FEATURE_BYTES_MESSAGE)));
```

It is required for webpush

Reply:
yes, but I don't use webpush since telegram server still uses the draft encoding (https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-encryption-encoding-03) that sends the keys in headers and it requires a gateway + modifications of the code that complicates stuff a lot so I choose to just use SIMPLE push https://core.telegram.org/api/push-updates that is ONLY a put with version=NUMBER and so FEATURE_BYTE_MESSAGE is not needed

Reply:
Ok, makes sens. All the distributors use the feature atm